frontend/src/pages/modelRegistry/screens/utils.ts

```diff
@@ -79,30 +79,46 @@ export const filterModelVersions = (
   unfilteredModelVersions: ModelVersion[],
   search: string,
   searchType: SearchType,
-): ModelVersion[] =>
-  unfilteredModelVersions.filter((mv: ModelVersion) => {
+): ModelVersion[] => {
+  const searchLower = search.toLowerCase();
+
+  return unfilteredModelVersions.filter((mv: ModelVersion) => {
     if (!search) {
       return true;
     }
 
     switch (searchType) {
-      case SearchType.KEYWORD:
-        return (
-          mv.name.toLowerCase().includes(search.toLowerCase()) ||
-          (mv.description && mv.description.toLowerCase().includes(search.toLowerCase()))
-        );
+      case SearchType.KEYWORD: {
+        const nameMatch = mv.name.toLowerCase().includes(searchLower);
+        const descriptionMatch =
+          mv.description && mv.description.toLowerCase().includes(searchLower);
 
-      case SearchType.AUTHOR:
+        let labelMatch = false;
+        labelMatch = Object.keys(mv.customProperties).some((key) => {
```

Comment:
this will get all customProperties which include properties, so although you don't see properties on the list view, you'll be able to filter on them which i don't think we want.

you could filter out anything where the `value.string_value` is truthy, but you may be better off using the `getLabels` function above which will return a string array... so something like

`getLabels(mv.customProperties).some( label => label.toLowerCase().includes(searchLower)`

also, rather than creating 3 vars here, you could just check them in a way that would shortcircuit if it finds a match.. like

        return (
          mv.name.toLowerCase().includes(searchLower) ||
          (mv.description && mv.description.toLowerCase().includes(searchLower)) ||
          getLabels(mv.customProperties).some( label => label.toLowerCase().includes(searchLower)
        );

i also wonder if parts of these two (filterModelVersions and filterRegisteredModels) can be factored out a bit. i think the only difference is author vs owner... so you could at least factor out the keyword search and accept `ModelVersion | RegisteredModel` or even potentially just have a more generic filter function that would take both and include both the `SearchType.OWNER` and `SearchType.AUTHOR` cases.

Reply:
@YuliaKrimerman I think we should consider doing this refactor @gitdallas is describing where we combine `filterRegisteredModels` and `filterModelVersions` using generics, but we can think about doing that in a followup PR since this one is blocking your other branch. I'll include that on a list of small tech debt items I plan to open shortly (there are some other things that came from the arch review meeting with @andrewballantyne).